instructor/common.py

```diff
@@ -583,32 +583,40 @@ def western_blot_lysate_type(request):
         },
         context_instance=RequestContext(request))
 
+@login_required
+def western_blot_antibody(request):
+    pk = request.session['assignment_id']
+    assignment = get_object_or_404(models.Assignment, id=pk)
+    wb, created = models.WesternBlot.objects.get_or_create(assignment=assignment)
 
-def western_blot_antibody_edit(request, assignment):
-    a = models.Assignment.objects.get(id=assignment)
-    wb, created = models.WesternBlot.objects.get_or_create(assignment=a)
-    wb.save()
-    WesternBlotAntibodyFormset = modelformset_factory(models.WesternBlotAntibody, extra=1, can_delete=True,
-                                                      exclude=['western_blot'])
-    message = ''
+    extra_fields = 0
+    if 'add' in request.POST or not models.WesternBlotAntibody.objects.filter(western_blot=wb):
```

Comment:
If you have something in the last row and you delete it, no fields show up. This isn't terribly important since the user can just click 'Add' to bring back the field, but it differs in behavior from the strains treatments page. If you calculate `extra_fields` after the delete I think you would avoid this problem